pkg/cloudprovider/providers/openstack/openstack_volumes.go

```diff
@@ -600,6 +640,32 @@ func (os *OpenStack) DisksAreAttached(instanceID string, volumeIDs []string) (ma
 	return attached, nil
 }
 
+// DisksAreAttachedByName queries if a list of volumes are attached to a compute instance by name
+func (os *OpenStack) DisksAreAttachedByName(nodeName types.NodeName, volumeIDs []string) (map[string]bool, error) {
+	attached := make(map[string]bool)
+	cClient, err := os.NewComputeV2()
+	if err != nil {
+		return attached, err
+	}
+	srv, err := getServerByName(cClient, nodeName, false)
+	if err != nil {
+		if err == ErrNotFound {
+			// instance not found anymore, mark all volumes as detached
+			for _, volumeID := range volumeIDs {
+				attached[volumeID] = false
+			}
+			return attached, nil
+		} else {
+			return attached, err
+		}
+	}
+	instanceID := "/" + srv.ID
+	if ind := strings.LastIndex(instanceID, "/"); ind >= 0 {
+		instanceID = instanceID[(ind + 1):]
+	}
+	return os.DisksAreAttached(instanceID, volumeIDs)
+}
+
```

Comment:
So this change is unrelated to issues this PR is supposed to fix right? I think this is fixing `VolumesAreAttached` mechanism but can we extract that into a separate PR? I feel as it is - we are fixing 2 bugs in this PR. This will be 3rd one.

Reply:
hmm, imo this should be here. Without this, if you shutdown node, it will mark all volumes as "available", but those are really attached to shutdowned node. This PR should fix those detach problems if node is shutdowned.